pkg/utils/resources/resources.go

```diff
@@ -93,20 +93,94 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
-func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
-	var resources v1.ResourceRequirements
+func PodRequests(pod *v1.Pod) v1.ResourceList {
+	reqs := v1.ResourceList{}
+
 	for _, container := range pod.Spec.Containers {
-		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
+		containerReqs := container.Resources.Requests
+
+		MergeInto(reqs, containerReqs)
+
 	}
+
+	restartableInitContainerReqs := v1.ResourceList{}
+	initContainerReqs := v1.ResourceList{}
+
 	for _, container := range pod.Spec.InitContainers {
-		resources.Requests = MaxResources(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = MaxResources(resources.Limits, container.Resources.Limits)
+		containerReqs := container.Resources.Requests
+
+		if container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
+			MergeInto(reqs, containerReqs)
+
+			MergeInto(restartableInitContainerReqs, containerReqs)
+			containerReqs = restartableInitContainerReqs
+		} else {
+			tmp := v1.ResourceList{}
+			MergeInto(tmp, containerReqs)
+			MergeInto(tmp, restartableInitContainerReqs)
+			containerReqs = tmp
```

Comment:
Same comment here, putting the `MaxReources` call here helps readability _a ton_. 

```go
maxInitContainerReqs = MaxResources(maxInitContainerReqs, Merge(restartableInitContainerReqs, container.Resource.Requests))
```